File: sdk/src/main/java/com/veertu/ankaMgmtSdk/AnkaNodeInfo.java
package com.veertu.ankaMgmtSdk;

import org.json.JSONObject;

public class AnkaNodeInfo {

  private final String ipAddress;
  private final String nodeId;
  private final String state;

  public AnkaNodeInfo(JSONObject jsonObject) {
    this.ipAddress = jsonObject.getString("ip_address");
    this.nodeId = jsonObject.getString("node_id");
    this.state = jsonObject.getString("state");
  }

  @Override
  public String toString() {
    return "AnkaNodeInfo{" +
      "ipAddress='" + ipAddress + '\'' +
      ", nodeId='" + nodeId + '\'' +
      ", state='" + state + '\'' +
      '}';
  }
}

File: sdk/src/main/kotlin/org/cirruslabs/anka/sdk/AnkaNodeInfo.kt
package org.cirruslabs.anka.sdk

import org.json.JSONObject
import kotlin.math.max

data class AnkaNodeInfo(
  val ipAddress: String,
  val nodeId: String,
  val state: String,
  val vmCount: Int,
  val capacity: Int
) {
  val hasCapacity: Boolean
    get() = vmCount < capacity

  val remainingCapacity: Int
    get() = max(0, capacity - vmCount)

  constructor(jsonObject: JSONObject) : this(
    jsonObject.getString("ip_address"),
    jsonObject.getString("node_id"),
    jsonObject.getString("state"),
    jsonObject.getInt("vm_count"),
    jsonObject.getInt("capacity")
  )
}
